Verbose first draft

CorylC opened the pull request:
Added Verbose function to be able to mute the save download message as well as the progress bar.

CorylC committed:
Verbose first draft

Added Verbose function to be able to mute the save download message as well as the progress bar.

ChuBL commented on openmindat/mindat_api.py:
```diff
@@ -185,24 +185,26 @@ def get_datetime(self):
         try:
             response = requests.get(url, headers=self._headers)
             new_results = response.json()['results']
             JSON_DATA["results"] += new_results
```
It is worth noting that the all-capital-letter naming for the parameters indicates that they are constant values. Please refer to [pep8](https://peps.python.org/pep-0008/#constants) for more details.
If you want to modify the constant dictionary, please consider using this format:
```
# Create a shallow copy of the constant dictionary
json_data = JSON_DATA.copy()
```

ChuBL reviewed (commented):
Everything with the verbose part looks good, except for modifying the constant parameter. The code should be functional regardless of the naming, but the naming style is important for further maintenance.

CorylC committed:
Added page_size catches

-added a catch for too long of a url
-added a catch for gateway timeout because of page_size being too large.
-small fix in geomaterials so id can be retrieved more consistently.

ChuBL commented on openmindat/mindat_api.py:
```diff
@@ -214,22 +214,36 @@ def get_mindat_json(self, PARAM_DICT, END_POINT, VERBOSE = 2):
+                # We will pass the response result directly
+                result_data = response_json
+                break
+            except ValueError:
```
The logic here seems weird. Did you mean the gateway timeout error will throw a `ValueError`? If so, I still recommend handling this exception differently: instead of wrapping the entire retrieval in a `while Ture` loop, use a for loop in range(3) to avoid adding `break`s everywhere. This might also remove the requirement of checking if `page_size`'s value is less than 200: raising an error when `page_size` is less than 200 appears weird. If I need clarification on the code, please let me know. Thanks.

ChuBL reviewed (commented):
See the comment in mindat_api.py

CorylC committed:
slight updates

-changed while loop to for
-decreased page_size cutoff to 150 from 200

CorylC committed:
float error fix

-ensured we are passing in int as page_size parameter after arithmetic.
-added a value error if the if statement terminates naturally.

CorylC committed:
removed unneeded sleep.

CorylC committed:
pbar initialize and for increase

-increased for loop for initial retrieval to 4 from 3
-initialized pbar when pbar isn't used to ensure no errors occur.
